File: LC_Gen_Codes/2810_no_setting.py
class Solution:
    def finalString(self, s: str) -> str:
        result = ""
        stack = []
        for c in s:
            if c == 'i':
                stack.reverse()
            else:
                stack.append(c)
        result += ''.join(stack)
        return result

# Testing the code
solution = Solution()
print(solution.finalString("string"))  # Output: "rtsng"
print(solution.finalString("poiinter"))  # Output: "ponter"

File: LC_Gen_Codes_Java/2810_no_setting.java
class Solution {
    public String finalString(String s) {
        StringBuilder result = new StringBuilder();
        StringBuilder stack = new StringBuilder();

        for (char c : s.toCharArray()) {
            if (c == 'i') {
                stack.reverse();
            } else {
                stack.append(c);
            }
        }

        result.append(stack);
        return result.toString();
    }
}